client/cmd/dexc-desktop/main.go

```diff
@@ -0,0 +1,303 @@
+
+| CLI version                       | Desktop version                          |
+|-----------------------------------|------------------------------------------|
+| Installed by building from source | Installed with an installer program.     |
+| or downloading a binary.          | Debian archive for Debian Linux,         |
+|                                   | Inno Setup for Windows.                  |
+|-----------------------------------|------------------------------------------|
+| Started by command-line.          | Started by selecting from the start/main |
+|                                   | menu, or by selecting a desktop icon or  |
+|                                   | pinned taskbar icon. CLI is fine too.    |
+|                                   | Program is installed in PATH.            |
+|-----------------------------------|------------------------------------------|
+| Accessed by going to localhost    | Opens in WebView, a simple window        |
+| address in the browser.           | backed by a web engine.                  |
+|-----------------------------------|------------------------------------------|
+| Shutdown via ctrl-c signal.       | When user closes window, continues       |
+| Prompt user to force shutdown if  | running in the background if there are   |
+| there are active orders.          | active orders. Run a little server that  |
+|                                   | synchronizes at start-up, enabling the   |
+|                                   | window to be reopened when the user      |
+|                                   | tries to start another instance.         |
+|-----------------------------------|------------------------------------------|
+
+Both versions use the same default client configuration file locations at
+AppDataDir("dexc").
+
+Since the program continues running in the background if there are active
+orders, there becomes a question of how and when to shutdown, or what
+happens when the user simply shuts off their computer, or it automatically restarts
+after updating.
+ 1) If there are no active orders when the user closes the window, dexc will
+    exit immediately.
+ 2) If we receive a SIGTERM signal, expected for system shutdown, shut down
+    immediately. Ctrl-c still works if running via CLI, with no prompt.
+ 3) If the window remains closed, but the active orders all resolve, shut down.
+    We check every minute while the window is closed.
+ 4) The user can kill the background program with a command-line argument,
+    --kill, which uses the sync server in the background to issue the command.
+
+DRAFT NOTES:
+
+Should we show a system-tray icon when running in the background?
+
+I (Buck) think we should offer a way for the user to run dexc as a system
+service (under the user, runs at login). The service would start with no window,
+but the UX would be unaffected, other than always being synced. This would
```

Comment:
A system service is a not an obvious yes IMO.  Without logging in, would it start up the native wallets and try to connect to external ones?  What about server connections?  It's quite likely that the user wants to control the lifetime of dex.  I know I do.  e.g. First start a VPN or the tor service, or even just not start until you are ready.

Also, in principle, I dislike apps with UI that require a service that hides in the background.  I think that if the UI goes away, the backend process should die unless there's some UI element remaining (like a tray icon).  The CLI case is a bit of an exception because we have a terminal opened if not built with systray support.

Reply:
I guess I'm imagining something like this...

1. On shutdown or before, store the refund transaction of any un-redeemed swaps, along with an estimate of a safe time to broadcast.
2. On startup, run "the service" to look in storage for any outstanding refunds. If there are, monitor the situation until either a) the refund can be broadcast, or b) the user logs in

It doesn't have to be the full dexc stack. It could just be a little utility function. It would need to be able to be run without logging in to **core**, so any wallet use would be precluded anyway. 

In my vision, the service would just remain running, open up a sync server, and maybe start syncing chains where it can. But that part seems like a no go for you. Understandable.

Reply:
That all makes sense, thanks for expanding.

The service thing is a personal preference clearly, and I'd not impose mine on everyone...  as long as an entirely non-daemonized app or app stack is an option.